### DISCUSSION TITLE
How to use Android UI with GameActivity?

### DISCUSSION THREAD

**Discussion by krm441** [android/ndk-samples#953] [Q&A]
It is more or less straightforward process in NDK using JNI to connect all the Kotlin buttons to their callbacks in the C++ code. But it seems not working in GameActivity. I can see that in the 'EndlessTunnel' project they have designed their own library for UI. But we still need Android UI (Jetpack for example). I understand that GameActivity renders to Surface and it has higher Z index - but what is the workaround?

**Comment by DanAlbert** [comment]
GameActivity questions will need to go to the Games SDK folks, because we don't know. https://source.android.com/docs/setup/contribute/report-bugs, use the "Games SDK" link.
